pkg/kubelet/config/common.go

```diff
@@ -120,6 +120,10 @@ func tryDecodeSinglePod(data []byte, defaultFn defaultFunc) (parsed bool, pod *v
 		return false, pod, fmt.Errorf("invalid pod: %#v", obj)
 	}
 
+	if newPod.Name == "" {
+		return true, pod, fmt.Errorf("invalid pod: name is needed for the pod")
+	}
+
```

Comment:
There is also tryDecodePodList function below, which is not calling this one.
What about to do this check in the `validation.ValidatePodCreate`? It's called from all places as far as I can see.

Reply:
> There is also tryDecodePodList function below, which is not calling this one.

Yeah, sorry I missed that.

> What about to do this check in the `validation.ValidatePodCreate`? It's called from all places as far as I can see.

Since `defaultFn` will be executed before `validation.ValidatePodCreate` and `pod.Name` will be assigned a value in `defaultFn`, I think it still has to be placed before `defaultFn`?

Reply:
Correct.
Let's start from adding the same check to the `tryDecodePodList` then.